### PR TITLE
WINC-555: [test] Use Windows Server 20H2 for Azure e2e tests

### PR DESCRIPTION
**Pull request by aravindhp**
- Use the corresponding container image for the deployments
- Use the appropriate PowerShell executable based on container image

### CODE REVIEW

**Comment by aravindhp** [comment]
/retest
```
 {"component":"entrypoint","file":"prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 2h0m0s timeout","severity":"error","time":"2021-02-24T01:13:52Z"}
{"component":"entrypoint","file":"prow/entrypoint/run.go:255","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 15s grace period","severity":"error","time":"2021-02-24T01:14:07Z"}
{"component":"entrypoint","error":"process timed out","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-02-24T01:14:07Z"}
error: failed to execute wrapped command: exit status 127 
```

**Comment by aravindhp** [comment]
/retest

Azure cluster failed to come up

**Comment by mansikulkarni96** [comment]
Thanks for working on this @aravindhp 
/lgtm

**Comment by aravindhp** [comment]
/hold
found an issue with the command (pwsh.exe vs powershell.exe) being used in the deployment. Puzzling how the test passed for me locally. I must have missed something.

**Comment by aravindhp** [comment]
>Puzzling how the test passed for me locally. I must have missed something.

I had missed pushing a local change.

**Comment by aravindhp** [comment]
/test azure-e2e-operator

`error: failed to load configuration: got unexpected http 504 status code from configresolver: <html><body><h1>504 Gateway Time-out</h1>`

File: test/e2e/network_test.go [hunk 1]
```diff
@@ -299,8 +300,13 @@ func (tc *testContext) deleteNamespace(name string) error {
 
 // deployWindowsWebServer creates a deployment with a single Windows Server pod, listening on port 80
 func (tc *testContext) deployWindowsWebServer(name string, affinity *v1.Affinity) (*appsv1.Deployment, error) {
+	powerShellExe := "pwsh.exe"
```
Comment (VaishnaviHire) [inline test/e2e/network_test.go:303]: Will this change be required [here](https://github.com/openshift/windows-machine-config-operator/pull/321/files#diff-84675b994e150b2fe4128b3f0a171854d1060ba909b9ff0aeb8833c8d9f4dae9R491) as well?

Comment (aravindhp) [reply test/e2e/network_test.go:303]: I don't follow. [here](https://github.com/openshift/windows-machine-config-operator/pull/321/files#diff-84675b994e150b2fe4128b3f0a171854d1060ba909b9ff0aeb8833c8d9f4dae9R491) takes me back this same line.

Comment (aravindhp) [reply test/e2e/network_test.go:303]: I think you are asking about the `getWinCurlerCommand`. Yes, it does.

**Comment by VaishnaviHire** [comment]
Thanks for the fix @aravindhp 

/lgtm

**Pull request merged**